### DISCUSSION TITLE
Export DatePickerInput

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI-react#758] [Ideas]
I started writing a DateInput using iTwin-react components. Then I realized it is very similar to the date input in the table filters. Would we be able to export the `DatePickerInput` component for use in consuming applications?

**Comment by veekeys** [comment]
Hmm.. I do not really remember what was the concern back then, that we did not want to make it generic.
Do you want to contribute for this one? It might be pretty straight forward (just not sure if its separate component or a part of `Input`).

**Reply by mayank99** [reply]
The concern is that the Table `DatePickerInput` is a very naive implementation that is not fit for general use (that's why someone else also did a custom implementation: https://github.com/iTwin/iTwinUI-react/discussions/674).

We would have to discuss the public API a bit more, and decide on questions like whether it should be a variant of `<Input>` or a new component, as well as do some research on other libraries' implementations. For proper keyboard support, it will not be a straightforward task. Check out this blog post to see the amount of work that goes into this component: https://react-spectrum.adobe.com/blog/date-and-time-pickers-for-all.html
